web/src/main.js

```diff
@@ -1280,6 +1287,13 @@ observe(state => state.application, (application) => {
   HistoryState.updateHistoryFromState(application);
 });
 
+// Observe for datetimechanes
+observe(state => state.data.grid, (grid) => {
```

Comment:
Use `state => (state.data.grid || {}).datetime`, then you don't need the `if` after, and can directly put the `setLastUpdated` as handler

Reply:
Using the if statement is necessary here as `setLastUpdated` is impure due to a direct reference to `getState().data.grid.datetime`.  The above suggestion doesn't solve accessing a property on a object that does not exist when within `setLastupdated`.

The only way to improve on this would be changing `setLastUpdated` to accept a param which would be `getState().data.grid.datetime`

So `observe` would look like
````javascript
observe(state => (state.data.grid || {}).datetime, (grid) => {
  setLastUpdated(grid.datetime);
});
````
And `setInterval` would be:
````javascript
setInterval(() => setLastUpdated(getState().data.grid.datetime), 60000);
````
IMO I would probably pass `state.application` as a parameter too, and add some default parameters to `setLastUpdated`.

The only problem is that the text that we set which is `moment(currentMoment).fromNow()`
will say _"a few seconds ago"_ if `currentmoment` is an empty string or undefined which is not true.

Reply:
Indeed you are right!